src/go/rpk/pkg/config/params.go

```diff
@@ -546,8 +546,34 @@ func (c *Config) addUnsetDefaults() {
 		brokers = []string{"127.0.0.1:9092"}
 	}
 
-	if len(r.AdminAPI.Addresses) == 0 {
-		r.AdminAPI.Addresses = []string{"127.0.0.1:9644"}
+	addrs := r.AdminAPI.Addresses
+	defer func() { r.AdminAPI.Addresses = addrs }()
+	if len(addrs) == 0 && len(c.Redpanda.AdminAPI) > 0 {
+		// We want to order the admin API addresses by:
+		// localhost -> loobpack -> private -> public.
+		var localhost, loopback, private, public []string
```

Comment:
The address order looks right, I would recommend also ordering non-mTLS endpoints first